### PR TITLE
Add fee vaults on L2

### PR DESCRIPTION
**Pull request by LEAFERx**
# Goals of PR

See #239 for more details.

Core changes:

- Improvement of github workflow.
- Improvement of configurations.
- Improvement of geth lib.
- Fix e2e test.

### CODE REVIEW

File: contracts/src/libraries/FeeVault.sol [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {UUPSUpgradeable} from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+/// @title FeeVault
+/// @notice The FeeVault contract contains the basic logic for the various different vault contracts
+///         used to hold fee revenue generated by the L2 system.
+abstract contract FeeVault is Initializable, UUPSUpgradeable, OwnableUpgradeable {
+    /// @notice Minimum balance before a withdrawal can be triggered.
+    uint256 public minWithdrawalAmount;
```
Comment (LEAFERx) [inline contracts/src/libraries/FeeVault.sol:13]: do we want this?

File: Makefile [hunk 2]
```diff
@@ -25,7 +25,7 @@ OPS_BINDINGS_TARGET = $(OPS_DIR)/bindings
 #CLEF_TARGET = $(SIDECAR_BIN)/clef
 
 install: geth magi sidecar ops
-geth: $(GETH_BIN_TARGET)
+geth: ops-bindings $(GETH_BIN_TARGET)
```
Comment (ujvl) [inline Makefile:28]: why is this necessary for geth

Comment (LEAFERx) [reply Makefile:28]: the hook needs the binding

File: contracts/src/libraries/FeeVault.sol (hunk 1 above)
Comment (ujvl) [reply contracts/src/libraries/FeeVault.sol:13]: sure it's fine, just configure it to 0 for now

Comment (LEAFERx) [reply contracts/src/libraries/FeeVault.sol:13]: it is configurable in `genesis_config.json` now

**Review by ujvl — APPROVED** [review]

File: config/local_devnet/genesis_config.json [hunk 3]
```diff
@@ -5,8 +5,12 @@
     "l2GenesisBlockDifficulty": "0x0",
     "l2GenesisBlockNumber": "0x0",
     "l2GenesisBlockGasUsed": "0x0",
-    "l2GenesisBlockBaseFeePerGas": "0x0",
+    "l2GenesisBlockBaseFeePerGas": "0x1",
     "l2GenesisBlockExtraData": "0x",
     "l2PredeployOwner": "0x000000000000000000000000000000000000dEaD",
+    "l2FeesWithdrawalAddress": "0x000000000000000000000000000000000000dEaD",
```
Comment (siosw) [inline config/local_devnet/genesis_config.json:11]: does having this in the genesis mean we can’t change the withdrawal address without a hard fork?

**Review by siosw — APPROVED** [review]

File: config/local_devnet/genesis_config.json (hunk 3 above)
Comment (ujvl) [reply config/local_devnet/genesis_config.json:11]: yeah (also this value is ofc temporary, we need to set it properly)
